manifests/server/slapdconf.pp

```diff
@@ -29,23 +29,36 @@
   }
 
   if $::openldap::server::ssl_cert {
-    if $::openldap::server::ssl_key {
-      validate_absolute_path($::openldap::server::ssl_cert)
-      validate_absolute_path($::openldap::server::ssl_key)
-      openldap::server::globalconf { 'TLSCertificate':
-        value => {
-          'TLSCertificateFile'    => $::openldap::server::ssl_cert,
-          'TLSCertificateKeyFile' => $::openldap::server::ssl_key,
-        },
-      }
-      if $::openldap::server::ssl_ca {
-        validate_absolute_path($::openldap::server::ssl_ca)
-        openldap::server::globalconf { 'TLSCACertificateFile':
-          value => $::openldap::server::ssl_ca,
+    if $::osfamily == 'RedHat' and versioncmp($::operatingsystemmajrelease, '6') > -1 {
+        validate_string($::openldap::server::ssl_cert)
+        openldap::server::globalconf { 'TLSCertificateFile':
+          value => $::openldap::server::ssl_cert,
+        }
+        if $::openldap::server::ssl_key {
+          validate_string($::openldap::server::ssl_key)
+          openldap::server::globalconf { 'TLSCertificateKeyFile':
+            value => $::openldap::server::ssl_key,
+          }
         }
-      }
     } else {
-      fail 'You must specify a ssl_key'
+      if $::openldap::server::ssl_key {
+        validate_absolute_path($::openldap::server::ssl_cert)
+        validate_absolute_path($::openldap::server::ssl_key)
+        openldap::server::globalconf { 'TLSCertificateFile':
```

Comment:
IIRC you have to do it in one run, that we we grouped TLSCertificateFile with TLSCertificateKeyFile